src/utils/api/helper.ts

```diff
@@ -1054,6 +1063,98 @@ export const updateWallet = async (
   )();
 };
 
+export const proceedToPayment = async (
+  {
+    checkData,
+    cardData,
+  }: {
+    checkData: PaymentCheckData;
+    cardData: {
+      cvv: string;
+      pan: string;
+      expiryDate: string;
+      holderName: string;
+    };
+  },
+  onError: (e: string) => void,
+  onResponse: (authUrl: string) => void
+) => {
+  mixpanel.track(TRANSACTION_AUTH_INIT.value, {
+    EVENT_ID: TRANSACTION_AUTH_INIT.value,
+  });
+  await pipe(
+    TE.tryCatch(
+      () =>
+        apiPaymentEcommerceClient.requestTransactionAuthorization({
+          transactionId: checkData.idPayment,
+          body: {
+            amount: checkData.amount as any,
+            fee: getPspSelected().fee as any,
```

Comment:
`as any` type assertions are not necessary. You can (and should) use `$TYPE.decode` for autogenerated types to convert between arbitrary objects and typesafe objects that use io-ts types